client/app/dedicated/server/servers/servers.routing.js

```diff
@@ -0,0 +1,108 @@
+import _ from 'lodash';
+
+export default /* @ngInject */ ($stateProvider) => {
+  const name = 'app.dedicated.servers';
+
+  $stateProvider.state(name, {
+    url: '/configuration/servers?page&pageSize&sort&sortOrder&filter',
+    component: 'dedicatedServerServers',
+    translations: { value: ['.', '../'], format: 'json' },
+    params: {
+      page: {
+        value: '1',
+        squash: true,
+      },
+      pageSize: {
+        value: '10',
+        squash: true,
+      },
+      sort: {
+        value: 'name',
+        squash: true,
+      },
+      sortOrder: {
+        value: 'ASC',
+        squash: true,
+      },
+      filter: {
+        value: '[]',
+        squash: true,
+      },
+    },
+    resolve: {
+      filter: /* @ngInject */ $transition$ => $transition$.params().filter,
+      orderUrl: /* @ngInject */ User => User.getUrlOf('dedicatedOrder'),
+      getServerDashboardLink: /* @ngInject */ $state => server => $state.href('app.dedicated.server', { productId: server.name }),
+      dedicatedServers: /* @ngInject */ (iceberg, $stateParams) => {
+        const filters = JSON.parse($stateParams.filter);
+        let request = iceberg('/dedicated/server')
+          .query()
+          .expand('CachedObjectList-Pages')
+          .limit($stateParams.pageSize)
+          .offset($stateParams.page)
+          .sort($stateParams.sort, $stateParams.sortOrder);
+
+        const FILTER_OPERATORS = {
+          contains: 'like',
+          is: 'eq',
+          isAfter: 'gt',
+          isBefore: 'lt',
+          isNot: 'ne',
+          smaller: 'lt',
+          bigger: 'gt',
+          startsWith: 'like',
+          endsWith: 'like',
+        };
+
+        filters.forEach(({ field, comparator, reference }) => {
+          request = request.addFilter(
+            field,
+            _.get(FILTER_OPERATORS, comparator),
+            reference.map((val) => {
+              switch (comparator.toUpperCase()) {
+                case 'CONTAINS':
+                  return `%25${val}%25`;
+                case 'STARTSWITH':
+                  return `${val}%25`;
+                case 'ENDSWITH':
+                  return `%25${val}`;
+                default:
+                  return val;
+              }
+            }),
```

Comment:
Maybe these could be in shared `constants`/`service` as they will probably be reused in future features/refactor